applications/luci-app-sshtunnel/htdocs/luci-static/resources/view/sshtunnel/ssh_servers.js

```diff
@@ -135,13 +131,35 @@ return view.extend({
 	},
 });
 
-function _splitSshKeys(sshFiles) {
-	var sshKeys = {};
-	for (var i = 0; i < sshFiles.length; i++) {
-		var sshPubKeyName = sshFiles[i];
-		var sshKeyName = sshPubKeyName.substring(0, sshPubKeyName.length - 4);
-		sshKeys[sshKeyName] = '';
+function _findAllPossibleIdKeys(entries) {
```

Comment:
Yes, you right, if you have a .pub but the id file removed then you'll see it as it exists. This normally should never happen and I skipped the case to keep simplicity. Not a big deal as for me.
The main goal was to just show the id_dropbear.
I can't take your version because although it fixes this particular issue it doesn't collect id keys without a .pub.

So if you don't mind I would ask you to accept the current solution. TBH I don't think it worth our time. At least I'm now busy with making the Dropbear itself compatible with OpenSSH

Reply:
Something like this then:
```js
function _findAllPossibleIdKeys(entries) {
    const sshKeyNames = new Set();

    const pubKeyFiles = entries.filter(item => item.type === 'file' && item.name.endsWith('.pub'));
    pubKeyFiles.forEach(pubFile => {
        const sshKeyName = pubFile.name.slice(0, -4);
        const correspondingFileExists = entries.some(item => (
            item.type === 'file' && item.name === sshKeyName && item.name.startsWith('id_'))
        ));

        if (correspondingFileExists) {
            sshKeyNames.add(sshKeyName);
        }
    });

    const sshKeyFiles = entries.filter(item => item.type === 'file' && !item.name.endsWith('.pub'));
    sshKeyFiles.forEach(keyFile => {
        const correspondingFileExists = entries.some(item => (
            item.type === 'file' && item.name.startsWith('id_'))
        ));

        if (correspondingFileExists) {
            sshKeyNames.add(item.name);
        }
    });

    return Array.from(sshKeyNames);
}
```

Reply:
thank you, I changed the function to process the case.